### PR TITLE
Prune metrics: precision & recall 6/n

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

re-routing Precision and Recall metrics
Follow-up #6505

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
requires merging #6572

File: tests/deprecated_api/test_remove_1-4.py [hunk 1]
```diff
@@ -130,16 +130,6 @@ def test_v1_4_0_deprecated_metrics():
...
 
-    # Testing deprecation of class_reduction arg in the *new* precision
-    from pytorch_lightning.metrics.functional import precision
-    with pytest.deprecated_call(match='will be removed in v1.4'):
-        precision(torch.randint(0, 2, (10, )), torch.randint(0, 2, (10, )), class_reduction='micro')
-
-    # Testing deprecation of class_reduction arg in the *new* recall
-    from pytorch_lightning.metrics.functional import recall
-    with pytest.deprecated_call(match='will be removed in v1.4'):
-        recall(torch.randint(0, 2, (10, )), torch.randint(0, 2, (10, )), class_reduction='micro')
```
Comment (Borda) [inline tests/deprecated_api/test_remove_1-4.py:141]: is the only above scope, that with this routing we drop deprecated `class_reduction` already in 1.3, not 1.4
 but I feel it is fine as we simplify the codebase...

Comment (SkafteNicki) [reply tests/deprecated_api/test_remove_1-4.py:141]: i agree with this :]

**Review by SkafteNicki — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by SeanNaren — APPROVED** [review]

**Review by teddykoker — APPROVED** [review]

**Pull request merged**